hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,231 @@
+*   Initialize flash scrambling and programming any excluded regions based on OTP or other information.
+
+There may be additional steps depending on flash design and software needs.
+
+### Power-Down Handling
+
+When the device enters deep sleep, the flash is automatically put into power down mode by the AST.
+When the device exits low power state, the flash is automatically powered up and it is boot ROM responsibility to poll for flash power-up complete.
+
+
+### Flash Brownout Handling
+
+When the supply of the device dips below a certain point during a stateful flash operation (program or erase) the flash requires the operation to terminate in a pre-defined manner.
+This sequence will be exclusively handled by the AST.
+The power manager is unaware of the difference between POR and flash brownout.
+Because of this, the software also cannot distinguish between these two reset causes.
+
+
+## Supported Low Power Modes
+
+This section details the various low power modes supported by OpenTitan.
+
+
+### Deep Sleep or Standby
+
+This represents the lowest power mode of the device (outside of full power down or device held in reset).
+During this state, the following states of the device shall be true
+
+*   All clocks other than the KHz slow clock shall be turned off at the source.
+*   All non-always-on digital domains are powered off
+*   I/O power domains may or may not be off
+    *   The state of the IO power domain shall have no impact on the digital core’s power budget
+
+
+### Normal Sleep
+
+This represents a fast low power mode of the device that compromises between power consumed and resume latency.
+During this state, the following states of the device shall be true
+
+*   All clocks other than the KHz slow clock shall be turned off at the source (AST)
+*   All non-always-on digital domains are kept on for fast resume
+*   Sensor countermeasures can be on opportunistically
+*   I/O power domains may or may not be off
+    *   The state of the IO power domain shall have no impact on the digital core’s power budget
+
+## Life Cycle and Debug
+
+This item is TBD
+
+Life cycle and TAP debug are two functions that may need to prevent the device from going into a low power state (imagine software is running and a TAP connection is made for specific debug).
+In these scenarios, it must be possible for the life cycle and debug functions to set a bit that prevents the power controller from entering sleep even if instructed to do so by software.
+The life cycle controller can maintain one such bit that is accessible from the life cycle TAP, while the normal debug TAP should maintain something similar.
+
+# Programmers Guide
+
+Please see [power management programmers model](https://docs.google.com/document/d/1w86rmvylJgZVmmQ6Q1YBcCp2VFctkQT3zJ408SJMLPE/edit?usp=sharing) for now.
+
+## Register Table
+
+{{< registers "hw/top_earlgrey/ip/pwrmgr/data/autogen/pwrmgr.hjson" >}}
```

Comment:
@eunchan 
@msfschaffner 
@gkelly 

do you guys know of a better way for this to be handled? It feels like the documentation we generate should be to a specific top, and in the documentation we should be able to just say `TOP_VERSION` or something. 

What do you guys think?

Reply:
I think that's ideal, because right now we usually have to hunt for the configuration that blocks are instantiated with inside of the toplevel when trying to set things up. I think the issue is that we need to make that generation happen at the `build_docs.py` level, and then populate it down through the documentation.